src/L2ERC20.sol

```diff
@@ -44,4 +47,15 @@ contract L2ERC20 is ERC20Votes, WormholeReceiver {
     require(clock() == L1_BLOCK.number(), "ERC20Votes: broken clock mode");
     return "mode=blocknumber&from=eip155:1";
   }
+
+  /// @notice Burn L2 tokens and unlock tokens on the L1.
+  /// @param account The account where the tokens will be transferred.
+  /// @param amount The amount of tokens to be unlocked.
+  function l1Unlock(address account, uint256 amount) external returns (uint256 sequence) {
+    _burn(account, amount);
```

Comment:
I think this should be `msg.sender` otherwise anyone can call this for any L2 holder 😬

Reply:
https://github.com/ScopeLift/l2-flexible-voting/pull/10/commits/ad70b0db9ad45595bd5f879b0e5489dde17a9ce4

Reply:
I don't think the change is what we want. Correct me if I'm wrong but my understanding is:

`account` is the address where the tokens will be sent on L1
`msg.sender` is the L2 token holder that is bridging them back to L1

Often they'll be the same, but not always. So it's ok for `account` to be something other than `msg.sender`. But the `_burn` call should use `msg.sender`, while the message passed via the bridge should use `account`

Reply:
Ah, I see now, yes that makes sense

Reply:
https://github.com/ScopeLift/l2-flexible-voting/pull/10/commits/546c9f6d77ad36f09f1fa87790c21b73756c7628